tools/MakeNiiFilesBIDSCompliant.pl

```diff
@@ -0,0 +1,705 @@
+                $headerNameMINC = $headerNameMINCArr[$j];
+                $headerName   = $headerNameArr[$j];
+                $headerName   =~ s/^\"+|\"$//g;
+                print "Adding now $headerName header to $headerFile\n" if $verbose;
+
+                $headerVal    =   &fetchMincHeader($mincFileName,$headerNameMINC);
+                # Some headers need to be explicitely converted to floats in Perl
+                # so json_encode does not add the double quotation around them
+                if (($headerNameMINC eq 'acquisition:repetition_time') ||
+                    ($headerNameMINC eq 'acquisition:echo_time') ||
+                    ($headerNameMINC eq 'acquisition:inversion_time') ||
+                    ($headerNameMINC eq 'dicom_0x0018:el_0x1314')) {
+                        $headerVal *= 1;
+                }
+
+
+#                if (defined($headerVal) && ($headerVal ne '')) {
+#                if (defined($headerVal) && ($headerVal != '')) {
+                if (defined($headerVal)) {
+                    $header_hash{$headerName} = $headerVal;
+                        print "     $headerName was found for $mincFileName with value $headerVal\n" if $verbose;
+
+                    # If scanner is Philips, store this as condition 1 being met
+                    if ($headerNameMINC eq 'study:manufacturer' && $headerVal =~ /Philips/i) {
+                        $manufacturerPhilips = 1;
+                    }
+                }
+                else {
+                    print "     $headerName was not found for $mincFileName\n" if $verbose;
+                }
+            }
+            
+            # If manufacturer is Philips, then add SliceOrder to the JSON manually
+            ######## This is just for the BETA version #########
+            ## See the TODO section for improvements needed in the future on SliceOrder ##
+            if ($manufacturerPhilips == 1) {
+                $extraHeader = "SliceOrder";
+                $extraHeader =~ s/^\"+|\"$//g;
+                if ($sliceOrderPhilips) {
+                    $extraHeaderVal = $sliceOrderPhilips;
+                }
+                else {
+                    print "   This is a Philips Scanner with no $extraHeader
+                    defined at the command line argument 'slice_order_philips'.
+                    Logging in the JSON as 'Not Supplied' \n" if $verbose;
+                }
+                $header_hash{$extraHeader} = $extraHeaderVal;
+                    print "    $extraHeaderVal was added for Philips Scanners'
+                    $extraHeader \n" if $verbose;
+            } else {
+                # get the SliceTiming from the proper header
+                # split on the ',', remove trailing '.' if exists, and add [] to make it a list
+                $headerNameMINC = 'dicom_0x0019:el_0x1029';
+                $extraHeader    = "SliceTiming";
+                $headerVal      =  &fetchMincHeader($mincFileName,$headerNameMINC);
```

Comment:
@MounaSafiHarab why get this from the MINC header while it is stored in the database?